core/src/main/java/org/stellar/anchor/sep31/Sep31Service.java

```diff
@@ -370,6 +427,27 @@ static Sep31InfoResponse createFromAssets(List<AssetInfo> assetInfos) {
     return response;
   }
 
+  @Data
+  public static class Context {
+    Sep31Transaction transaction;
+    Sep31PostTransactionRequest request;
+    JwtToken jwtToken;
+    Amount fee;
+    AssetInfo asset;
+    static ThreadLocal<Context> context = new ThreadLocal<>();
+
+    public static Context get() {
+      if (context.get() == null) {
+        context.set(new Context());
+      }
+      return context.get();
+    }
+
+    public static void reset() {
+      context.set(null);
+    }
+  }
+
```

Comment:
❗ I don't think this is thread safe. What if we're processing multiple transaction contexts at the same time?

Instead of using a static context variable, we should use different instances of context every time we need to process a different transaction.

Reply:
I see a few problems with this approach:
* This works like a singleton, which is an anti-pattern.
* The name `Context` is very broad, we should use something more specific.
* This class handles POST, PATCH, and GET but all of them have access to this static singleton Context, meaning any request can interfere with the Context values. Multiple POST requests running in parallel certainly will interfere with each other and cause unpredictable results.

My suggestion is to actually create a local instance of this `PostTxContext` and pass it around to the private methods that need access to it. Or even better: only pass over the values that need to be used. For instance, if a `validateSenderAndReceiver` method just needs the sender and receiver you should only pass these properties, this will make the code more predictable and easier to write tests for it.

Reply:
This is actually thread-safe. 😄 Line #437 is an object of ThreadLocal. ([Some intro here](https://www.baeldung.com/java-threadlocal)). It is designed to store context of a thread execution.

1. The singleton is the ThreadLocal and the objects are per-thread. 😃 
2. The class name `Context` is nested in Sep31Service. To access the class, the name would be Sep31Service.Context. 
3. This should be the same as #1. 

I will be happy to call if it is not explained clearly.